automatically build and publish Debian packages at https://gitlab.com/nodiscc/pidgin-opensteamworks/-/releases

nodiscc opened the pull request:
- https://gitlab.com/nodiscc/pidgin-opensteamworks/ configured to mirror https://github.com/EionRobb/pidgin-opensteamworks
- let gitlab CI build packages for debian stretch/buster on every git tag and upload them to the package registry (https://gitlab.com/nodiscc/pidgin-opensteamworks/-/packages)
- let gitlab CI create a release at https://gitlab.com/nodiscc/pidgin-opensteamworks/-/releases and link to the packages download page
- run a lintian pass on resulting packages
- use debian buster-backports image as checkinstall is only available in buster-backports
- update documentation
- ref. https://docs.gitlab.com/ce/ci/yaml/README.html

![01](https://user-images.githubusercontent.com/3950390/122609543-b0229080-d06d-11eb-955d-6a1f99d35421.png)

    ---->

![02](https://user-images.githubusercontent.com/3950390/122609550-b3b61780-d06d-11eb-8b5b-1de13de88c84.png)

    ---->

![03](https://user-images.githubusercontent.com/3950390/122609559-b6187180-d06d-11eb-84e2-3f74a311e4b6.png)

    ---->

![04](https://user-images.githubusercontent.com/3950390/122609563-b87acb80-d06d-11eb-9b6d-aa0b69f1c4cf.png)

    ---->

![05](https://user-images.githubusercontent.com/3950390/122609569-bb75bc00-d06d-11eb-951b-493ec9a78c7e.png)

EionRobb commented:
Nice! That looks super useful :) and thank you for updating the changelog notes, that's above and beyond! 

Only comment I have is that the deb shouldn't depend on libsecret/gnome-keyring to run, that they're only recommended. They're needed at compiletime but not needed at runtime as they're dynamically loaded as necessary.

nodiscc commented:
I will remove the dependency on libsecret/gnome-keyring.

What I would really like is having the built `.deb` files directly on https://gitlab.com/nodiscc/pidgin-opensteamworks/-/releases, this way we could simply link to this releases page from the README. But it's not yet possible to upload build artifacts to a release (https://gitlab.com/gitlab-org/gitlab/-/issues/36133).

So I either:
 - Wait for the 3.6 release which will make this possible, and update `.gitlab-ci.yml` accordingly
 - Find a way to add direct links to the artifacts in the release description (it should be easy, the URL is `https://gitlab.com/nodiscc/pidgin-opensteamworks/-/jobs/artifacts/$CI_COMMIT_TAG/browse?job=package-debian-stretch`)

I think I will do the latter

EionRobb commented:
Any luck removing the libsecret depends?

nodiscc commented:
Hi, I have started working on this again (the 2 commits above are a mistake, they should have gone to a temporary work branch). I will remove libsecret and automate the upload of `.deb`s to https://gitlab.com/nodiscc/pidgin-opensteamworks/-/packages and let you know when everything is ready

nodiscc commented:
Rebased, Pull Request description updated. Once you merge this, feel free to add a `1.7.1` tag and push it, https://gitlab.com/nodiscc/pidgin-opensteamworks should automatically mirror the changes and the new tag and trigger a build/release/package upload. I added you as a maintainer of the repo

nodiscc commented:
@EionRobb can you tag `1.7.1`? Thanks
